unicodetools/src/main/java/org/unicode/text/utility/Utility.java

```diff
@@ -928,6 +930,34 @@ public static String join(long[] array, String divider) {
         "1.1.0",
     };
 
+    public static final List<VersionInfo> UNICODE_VERSIONS =
+            Arrays.asList(searchPath).stream()
+                    .map(VersionInfo::getInstance)
+                    .collect(Collectors.toList());
+
+    public static VersionInfo getVersionPreceding(VersionInfo version) {
+        for (VersionInfo candidate : UNICODE_VERSIONS) {
+            if (candidate.compareTo(version) < 0) {
+                return candidate;
+            }
+        }
+        return null;
+    }
+
+    public static VersionInfo getVersionFollowing(VersionInfo version) {
+        VersionInfo result = null;
+        for (VersionInfo candidate : UNICODE_VERSIONS) {
+            if (candidate.compareTo(version) > 0) {
+                return result = candidate;
```

Comment:
This looks wrong: It returns the first thing in the list if it's > candidate.
I think you want to remove the "return" here and just remember the last candidate > version going back in time.

Reply:
Ah yes, it looks like I forgot to remove the `return` (returning an assignment to a local variable definitely doesn’t look intentional!).